### ISSUE TITLE
getCompilerOptionsDiagnostics didn't report when use compiler API 

### ISSUE THREAD

**Issue by WanderWang** [Microsoft/TypeScript#6174]
here is a example .

```
// a.ts
var a = (<div>HelloWorld</div>
```

when I run ``` tsc a.ts ``` in terminal , it will report 

```
a.tsx(2,9): error TS17004: Cannot use JSX unless the '--jsx' flag is provided.
```

but when I compile it with compiler API ( from https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API ) , it will report nothing . ( by the way , seems that only ``` getCompilerOptionsDiagnostics ``` report nothing , other diagnositics report right ) 

here is the example code

```
function logErrors(fileName: string) {
        let allDiagnostics = services.getCompilerOptionsDiagnostics()
            .concat(services.getSyntacticDiagnostics(fileName))
            .concat(services.getSemanticDiagnostics(fileName));

        allDiagnostics.forEach(diagnostic => {
            let message = ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n");
            if (diagnostic.file) {
                let { line, character } = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
                console.log(`  Error ${diagnostic.file.fileName} (${line + 1},${character + 1}): ${message}`);
            }
            else {
                console.log(`  Error: ${message}`);
            }
        });
    }
```

**Comment by DanielRosenwasser** [comment]
Do you have complete code (or a minimal repro) that you could share so we can diagnose the issue?

**Comment by DanielRosenwasser** [comment]
@WanderWang any update on this?

**Comment by WanderWang** [comment]
@DanielRosenwasser    sorry . I will make a simple example tonight

**Comment by WanderWang** [comment]
" tonight " means that 10 hours later because of I'm in Beijing ,China

**Comment by WanderWang** [comment]
Hello @DanielRosenwasser 
Here is a simple example https://github.com/WanderWang/TypeScriptIssue6174

I got a mistake . When I haven't set ```jsx``` field to ``` compilerOptions``` , it will report  ```error TS17004: Cannot use JSX unless the '--jsx' flag is provided.```

But when I input a error jsx value ( such as '100' or 'react' ) ， it will not report a " Unknown jsx value " exception  so I thought that it 's right in a mistake.

Maybe you should check the compilerOption as a Defensive programming . But it 's out of range in this issue
